### PR TITLE
feat: one2one migrated to mls system message [WPB-6195]

### PR DESCRIPTION
**Pull request by PatrykBuniX**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-6195" title="WPB-6195" target="_blank">TaskWPB-6195</a>  [Web] Insert a system message after migrating 1:1 conversation
  </td></table>
  <br />
 

## Description

When Proteus 1:1 conversation gets migrated to MLS protocol, we should inform the user about it and inject a system message into the conversation.
Screenshot 2024-01-19 at 12 55 25

## Checklist

- [x] PR has been self reviewed by the author;
- [x] Hard-to-understand areas of the code have been commented;
- [x] If it is a core feature, unit tests have been added;

### CODE REVIEW

File: src/script/components/MessagesList/Message/SystemMessage/SystemMessage.tsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ import {JoinedAfterMLSMigrationFinalisationMessage} from 'src/script/entity/mess
 import {MessageTimerUpdateMessage} from 'src/script/entity/message/MessageTimerUpdateMessage';
 import {MLSConversationRecoveredMessage} from 'src/script/entity/message/MLSConversationRecoveredMessage';
 import {MLSMigrationFinalisationOngoingCallMessage} from 'src/script/entity/message/MLSMigrationFinalisationOngoingCallMessage';
+import {One2OneMigratedToMlsMessage} from 'src/script/entity/message/One2OneMigratedToMlsMessage';
```
Comment (przemvs) [inline src/script/components/MessagesList/Message/SystemMessage/SystemMessage.tsx]: Hmm.. I think we have different Event names OneToOne[...], so please rename to OneToOneMigratedToMLSMessage

File: src/script/conversation/ConversationRepository.ts [hunk 2]
```diff
@@ -1464,6 +1464,14 @@ export class ConversationRepository {
 
     ConversationMapper.updateProperties(mlsConversation, updates);
 
+    const wasProteus1to1ActiveConversation = proteusConversations.some(conversation =>
+      this.conversationState.isActiveConversation(conversation),
+    );
+
+    const wasProteusConnectionIncomingRequest = proteusConversations.some(
+      conversation => conversation.type() === CONVERSATION_TYPE.CONNECT,
+    );
```
Comment (przemvs) [inline src/script/conversation/ConversationRepository.ts]: We have ConversationSelectors, so maybe move this to ConversationSelector, isConnectConversation ? WDYT ?

File: src/script/conversation/ConversationRepository.ts [hunk 3]
```diff
@@ -1852,10 +1875,6 @@ export class ConversationRepository {
 
       conversation.connection(connectionEntity);
 
-      if (connectionEntity.isConnected()) {
-        conversation.type(CONVERSATION_TYPE.ONE_TO_ONE);
-      }
```
Comment (PatrykBuniX) [inline src/script/conversation/ConversationRepository.ts:1857]: This part was moved to `initProteus1to1Conversation` method - no need to run this on mls 1:1 - it's always of type `ONE_TO_ONE`.

File: src/script/conversation/ConversationRepository.ts [hunk 4]
```diff
@@ -1472,9 +1487,11 @@ export class ConversationRepository {
       }),
     );
 
-    const wasProteus1to1ActiveConversation =
-      !!proteusConversations &&
-      proteusConversations.some(conversation => this.conversationState.isActiveConversation(conversation));
+    // Because of the current architecture and the fact that we present a connection request as a conversation of connect type,
+    // we don't want to inject conversation migrated even if the only proteus 1:1 conversation we had was a connection request.
+    if (!wasProteusConnectionIncomingRequest) {
```
Comment (PatrykBuniX) [inline src/script/conversation/ConversationRepository.ts:1492]: We don't want the message to be injected if mls 1:1 was created right after accepting a connection (proteus 1:1 conversation was never used)

File: src/script/conversation/ConversationRepository.ts [hunk 5]
```diff
@@ -1472,9 +1487,11 @@ export class ConversationRepository {
       }),
     );
 
-    const wasProteus1to1ActiveConversation =
-      !!proteusConversations &&
-      proteusConversations.some(conversation => this.conversationState.isActiveConversation(conversation));
+    // Because of the current architecture and the fact that we present a connection request as a conversation of connect type,
+    // we don't want to inject conversation migrated even if the only proteus 1:1 conversation we had was a connection request.
```
Comment (atomrc) [inline src/script/conversation/ConversationRepository.ts]:
```suggestion
    // we don't want to inject conversation migrated event if the only proteus 1:1 conversation we had was a connection request.
```

**Review by atomrc — APPROVED** [review]

**Pull request merged**